tests/integration/tier0/config-file/test_config_file.py

```diff
@@ -0,0 +1,65 @@
+import os
+
+
+def test_config_file(convert2rhel):
+    paths = (
+        ("~/.convert2rhel_custom.ini", "[subscription_manager]\nactivation_key = config_activationkey"),
+        ("~/.convert2rhel.ini", "[subscription_manager]\npassword = config_password"),
+        ("~/password_file", "file_password"),
+    )
+
+    for path, content in paths:
+        f = open(os.path.expanduser(path), "w")
+        f.write(content)
+        f.close()
+        os.chmod(os.path.expanduser(path), 0o600)
+
+    with convert2rhel('--no-rpm-va --debug -c "~/.convert2rhel_custom.ini"') as c2r:
+        c2r.expect("DEBUG - Found activation_key in /root/.convert2rhel_custom.ini")
+        c2r.sendcontrol("c")
+    assert c2r.exitstatus != 0
+
+    with convert2rhel("--no-rpm-va --debug") as c2r:
+        c2r.expect("DEBUG - Found password in /root/.convert2rhel.ini")
+        c2r.sendcontrol("c")
+    assert c2r.exitstatus != 0
+
+    with convert2rhel(("--no-rpm-va --password password --debug")) as c2r:
+        c2r.expect("DEBUG - Found password in /root/.convert2rhel.ini")
+        c2r.expect("WARNING - Command line authentication method take precedence over method in configuration file.")
+        c2r.sendcontrol("c")
+    assert c2r.exitstatus != 0
+
+    with convert2rhel(('--no-rpm-va -f "~/password_file" --debug')) as c2r:
+        c2r.expect("DEBUG - Found password in /root/.convert2rhel.ini")
+        c2r.expect("WARNING - Deprecated. Use -c | --config-file instead.")
+        c2r.expect("WARNING - Password file take precedence over the config file.")
+        c2r.sendcontrol("c")
+    assert c2r.exitstatus != 0
+
+    # Create one more config file in lower priority location
+    f = open("/etc/convert2rhel.ini", "w")
+    f.write("[subscription_manager]\nactivation_key = config2_activationkey")
+    f.close()
+    os.chmod("/etc/convert2rhel.ini", 0o600)
+
+    with convert2rhel("--no-rpm-va --debug") as c2r:
+        c2r.expect("DEBUG - Found password in /root/.convert2rhel.ini")
+        c2r.expect("DEBUG - Found activation_key in /etc/convert2rhel.ini")
+        c2r.expect("WARNING - Set only one of password or activation key. Activation key take precedence.")
+        c2r.sendcontrol("c")
+    assert c2r.exitstatus != 0
+
+    # Change content of the file in lower priority location
+    f = open("/etc/convert2rhel.ini", "w")
+    f.write("[subscription_manager]\npassword = config_password")
+    f.close()
+
+    with convert2rhel("--no-rpm-va --debug") as c2r:
+        c2r.expect("DEBUG - Found password in /root/.convert2rhel.ini")
+        c2r.sendcontrol("c")
+    assert c2r.exitstatus != 0
+
+    for path, content in paths:
+        os.remove(os.path.expanduser(path))
+    os.remove(os.path.expanduser("/etc/convert2rhel.ini"))
```

Comment:
What do you think about splitting the integration tests into more tests in the same file? I feel that this can help in the readability and debugging process when needed. For starters, you could take all the occurrences of `with convert2rhel(...) as c2r` and split them into different tests. 

What do you think, @danmyway  and @kokesak?